compute/resource_job.go

```diff
@@ -92,6 +93,17 @@ var jobSchema = internal.StructToSchema(JobSettings{},
 			"`new_cluster` for greater reliability."
 		s["new_cluster"].Description = "Same set of parameters as for " +
 			"[databricks_cluster](cluster.md) resource."
+		if v, err := internal.SchemaPath(s, "new_cluster", "spark_conf"); err == nil {
+			v.DiffSuppressFunc = func(k, old, new string, d *schema.ResourceData) bool {
+				isPossiblyLegacyConfig := "new_cluster.0.spark_conf.%" == k && "1" == old && "0" == new
+				isLegacyConfig := "new_cluster.0.spark_conf.spark.databricks.delta.preview.enabled" == k
+				if isPossiblyLegacyConfig || isLegacyConfig {
+					log.Printf("[DEBUG] Suppressing diff for k=%#v old=%#v new=%#v", k, old, new)
+					return true
+				}
+				return false
```

Comment:
does it make sense to move it into shared function, because the difference is only that this function is different only in prefix `spark_conf.` vs `new_cluster.0.spark_conf.`

Reply:
hmm are we even sure if this is relevant to jobs? AFAIK when i create a job without spark configs that job does not have spark configs configured by default it only happens in the clusters api.

Reply:
I verified with the following create payload:

```json
{
   "name":"Nightly model training",
   "new_cluster":{
      "spark_version":"5.3.x-scala2.11",
      "node_type_id":"r3.xlarge",
      "aws_attributes":{
         "availability":"ON_DEMAND"
      },
      "num_workers":10
   },
   "libraries":[
      {
         "jar":"dbfs:/my-jar.jar"
      },
      {
         "maven":{
            "coordinates":"org.jsoup:jsoup:1.7.2"
         }
      }
   ],
   "email_notifications":{
      "on_start":[
         
      ],
      "on_success":[
         
      ],
      "on_failure":[
         
      ]
   },
   "timeout_seconds":3600,
   "max_retries":1,
   "schedule":{
      "quartz_cron_expression":"0 15 22 ? * *",
      "timezone_id":"America/Los_Angeles"
   },
   "spark_jar_task":{
      "main_class_name":"com.databricks.ComputeModels"
   }
}
```

The response from the GET:

```json
{
    "job_id": 4393,
    "settings": {
        "name": "Nightly model training",
        "new_cluster": {
            "spark_version": "5.3.x-scala2.11",
            "aws_attributes": {
                "zone_id": "us-west-2b",
                "availability": "ON_DEMAND"
            },
            "node_type_id": "r3.xlarge",
            "enable_elastic_disk": false,
            "num_workers": 10
        },
        "libraries": [
            {
                "jar": "dbfs:/my-jar.jar"
            },
            {
                "maven": {
                    "coordinates": "org.jsoup:jsoup:1.7.2"
                }
            }
        ],
        "email_notifications": {},
        "timeout_seconds": 3600,
        "max_retries": 1,
        "schedule": {
            "quartz_cron_expression": "0 15 22 ? * *",
            "timezone_id": "America/Los_Angeles",
            "pause_status": "UNPAUSED"
        },
        "spark_jar_task": {
            "jar_uri": "",
            "main_class_name": "com.databricks.ComputeModels",
            "run_as_repl": true
        },
        "max_concurrent_runs": 1
    },
    "created_time": 1600863327560,
    "creator_user_name": "sri.tikkireddy@databricks.com"
}
```

As you can see there is no conf auto added. Just the api for clusters being a bit bizarre 😄

Reply:
hmm i guess they are seeing different behavior in here: https://github.com/databrickslabs/terraform-provider-databricks/issues/323. Which is inconsistent to what i am seeing so then i guess we may need this but it is bizarre. 

And not repeatable.

Reply:
So because of azure, we have to suppress diffs :)